packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -20,6 +33,36 @@ def _ensure_routable():
     return _ensure_routable
 
 
+@retrying.retry(wait_fixed=2000,
```

Comment:
This retries on 
* AssertionErrors too.
* retries the tests too!

I guess, it was unintentional.  You might have to factor out `_check_overlay` outside of this test code and have a retry application only there (as it does the requests.get call).